### PR TITLE
MueLu: regionMG disable rebalance tests in serial build

### PR DESCRIPTION
**Pull request by pohm01**
@trilinos/muelu 

## Motivation

Fixes failing regionMG tests by removing them from serial builds. The removed tests were for rebalancing and thus not as relevant to a serial build.

## Testing
The failing tests no longer appear in serial builds on my machine, but still appear and pass on non-serial builds.

### CODE REVIEW

File: packages/muelu/research/regionMG/test/structured/CMakeLists.txt [hunk 1]
```diff
@@ -144,6 +145,7 @@ IF (${PACKAGE_NAME}_ENABLE_Tpetra AND ${PACKAGE_NAME}_ENABLE_Amesos2)
     COMM serial mpi
```
Comment (cgcgcg) [inline packages/muelu/research/regionMG/test/structured/CMakeLists.txt:145]: Wouldn't it be enough to remove `serial`?

Comment (pohm01) [reply packages/muelu/research/regionMG/test/structured/CMakeLists.txt:145]: Is that how this works? Because that does seem like it would be better.

Comment (cgcgcg) [reply packages/muelu/research/regionMG/test/structured/CMakeLists.txt:145]: Yes, in combination with `NUM_MPI_PROCS 4` this should do the trick

Comment (pohm01) [reply packages/muelu/research/regionMG/test/structured/CMakeLists.txt:145]: I will test that, then change this PR to be that if it works.

Comment (pohm01) [reply packages/muelu/research/regionMG/test/structured/CMakeLists.txt]: I did it.

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**